src/go/k8s/pkg/resources/configmap.go

```diff
@@ -817,7 +823,31 @@ func (r *ConfigMapResource) PrepareSeedServerList(cr *config.RedpandaNodeConfig)
 		}
 	}
 
-	for i := int32(0); i < replicas; i++ {
+	// Clamp the broker list in seed servers to be at most 3 entries. This
+	// gives us the best possible non-service discovery based solution.
+	//
+	// - Scaling up won't require existing brokers to be restarted to update
+	//   their configs.
+	// - Scaling down to any number > 3 won't require existing brokers to be
+	//   restarted to update their configs.
+	// - If scaling down to < 3, we'll eat the cost of a rolling restart as
+	//   this is VERY unlikely to actually occur.
+	// - 3 entries is reasonably stable for cluster formation.
+	// - The stability of STS pod names means the list will be stable across
+	//   the lifetime of the cluster.
+	//
+	// Ideally we'd be able to lean on K8s service discovery here but RP
+	// asserts that each hostname in the seed_servers is 1:1 with a given
+	// node_id.
+	// https://github.com/redpanda-data/redpanda/blob/5629704284ed6c289b3032c6650944d25dfa48f6/src/v/cluster/cluster_discovery.cc#L308
+	// There are also a slew of race conditions that may cause formation issues
+	// if using a naive SRV calculation approach.
+	// In the future, it may be feasible to have an init container that fetches
+	// custom resources and calculates the seed_broker list on demand.
+
+	const maxSeedServers = 3
+
+	for i := int32(0); i < replicas && i < maxSeedServers; i++ {
```

Comment:
> When scaling down, will the first replicas in this list be the ones that remain?

Yes, this list will always be replicas 0, 1, and 2. Thanks to the STS's ordering guarantees we can rely on them always being around.

> When doing a rolling upgrade of infrastructure (e.g., new K8s version), will this list be sufficient if the nodes are all going away?

The listed nodes will always exist (again STS ordinal system). It should be fine as long as nodes are migrated one at a time and this wouldn't really function any differently than what was previously present. Especially in the case of 3 node clusters.

> Is cr.EmptySeedStartsCluster = False?

That's handled on the line above and it looks like it's always set to false if it's a supported redpanda version.

> How does this work when there is multi-AZ?

I don't really see how that's relevant but it would function the same way it previously worked. This logic only capped the length of the list. The contents is otherwise computed the exact same way.